### DISCUSSION TITLE
Drag Drop onready shortcut Hint please

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#10105] [Editor]
### Problem: I always forget if it's shift or CTRL or ALT that does the onready line vs just the node name when dragging dropping a node from scene tree into the script window while holding CTRL/shift/ALT.

Right now the drag drop mouse "label" when dragging is just the name of the node.

Is it possible to make this drag drop mouse label show that it will make an onready script line if CTRL is being held while dragging?

I am not specifically sure what this improved hint label should be, Maybe "OR node name" instead of just "node name" if CTRL is held?

I just need something to tell me i'm holding the right key to do the right thing before I drop.

**Comment by Calinou** [comment]
The drag-and-drop preview is set by the Scene tree dock, so displaying a hint that you can hold <kbd>Ctrl</kbd> within this hint would be out of place (since it's not guaranteed you'll drop the node onto the script editor). It sounds difficult to show this hint only when currently hovering the script editor.

**Reply by shak2** [reply]
If you hold CTRL and drag drop into a script it does currently make the onready variable, so the code does somehow already recognise that the mouse is within the script window bounds when doing this shortcut.

If (CTRL held and mouse over script window) {prepend ON to label}

I still think this is a good idea as watching pro godot devs live streaming shows they regularly make this mistake too. It's not just beginners.

**Reply by Calinou** [reply]
> If you hold CTRL and drag drop into a script it does currently make the onready variable, so the code does somehow already recognise that the mouse is within the script window bounds when doing this shortcut.

Yes, the script editor does that when you *release* the mouse inside its bounds. It does not change the drag-and-drop *preview* before you release the mouse.

The drag-and-drop preview is not drawn by the script editor – it's drawn by Godot's GUI system at a higher level. What is being drawn is decided in the Scene tree dock. You could try to make a label follow alongside but this would be really hacky and hard to position correctly.
